src/common/templates/_monitoring.tpl

```diff
@@ -9,6 +9,16 @@
 {{- $port := (pluck "port" $monitoring) | first -}}
 ENABLE_PROMETHEUS_COLLECTOR: {{ default "false" $enabled | quote }}
 PROMETHEUS_COLLECTOR_PORT: {{ default "8889" $port | quote }}
+{{- $otelEnabled := default false (or ((.Values.monitoring).otel).enabled (((.Values.global).monitoring).otel).enabled) }}
```

Comment:
using braces avoids nil pointer exceptions.
ref: https://github.com/helm/helm/issues/8026#issuecomment-881216078

Reply:
using dig function is also an alternative.